packages/dev/core/src/Misc/arrayTools.ts

```diff
@@ -56,43 +55,51 @@ export class ArrayTools {
     }
 
     private static _ProxySet<T>(observable: Observable<INotifyArrayChangeType<T>>, target: Array<T>, property: string | symbol, value: T): boolean {
-        observable.notifyObservers({ target, values: [value], operation: "set" });
+        observable.notifyObservers({ target, previousLength: target.length });
         return Reflect.set(target, property, value);
     }
 
     private static _ProxyPushOrUnshift<T>(operation: "push" | "unshift", observable: Observable<INotifyArrayChangeType<T>>, target: Array<T>, ...values: Array<T>): number {
-        observable.notifyObservers({ target, values, operation });
+        observable.notifyObservers({ target, previousLength: target.length });

```

Comment:
Note that for the `renderList` array of the `RenderTargetTexture` class it is a change in behaviour: the **push** operation was done before the additional code was executed. Here, the additional code will be execute first (through the `notifyObservers` call) and the **push** operation will be done afterwards.

I don't know if it will break something or not, but it's for sure a different behaviour. To keep the existing behaviour it would be safer to notify the observers after the operation has been done.

Also, I think it should be wise to always notify the observers at the same point (either before or after the operation on the array has been performed) for the sake of clarity. Currently:
* set, push/unshift, delete: observers notified before the operation
* pop/shift, splice: observers notified after the operation

Or maybe it is on purpose? In that case, adding some doc on `MakeObservableArray` explaining when the observers are notified in each case would probably be a good idea.